rajawali/src/main/java/org/rajawali3d/view/TextureView.kt

```diff
@@ -0,0 +1,1537 @@
+            } else if (attr == R.styleable.TextureView_bitsDepth) {
+                bitsDepth = array.getInteger(attr, 16)
+            }
+        }
+        array.recycle()
+    }
+
+    private fun initialize() {
+        val glesMajorVersion = Capabilities.getGLESMajorVersion()
+        setEGLContextClientVersion(glesMajorVersion)
+
+        setEGLConfigChooser(RajawaliEGLConfigChooser(glesMajorVersion, antiAliasingConfig, multiSampleCount,
+                bitsRed, bitsGreen, bitsBlue, bitsAlpha, bitsDepth))
+    }
+
+    private fun checkRenderThreadState() {
+        if (glThread != null) {
+            throw IllegalStateException("setRenderer has already been called for this instance.")
+        }
+    }
+
+    /**
+     * This method is part of the SurfaceTexture.Callback interface, and is
+     * not normally called or subclassed by clients of TextureView.
+     */
+    private fun surfaceCreated(width: Int, height: Int) {
+        glThread!!.surfaceCreated(width, height)
+    }
+
+    /**
+     * This method is part of the SurfaceTexture.Callback interface, and is
+     * not normally called or subclassed by clients of TextureView.
+     */
+    private fun surfaceDestroyed() {
+        // Surface will be destroyed when we return
+        glThread!!.surfaceDestroyed()
+    }
+
+    /**
+     * This method is part of the SurfaceTexture.Callback interface, and is
+     * not normally called or subclassed by clients of TextureView.
+     */
+    private fun surfaceChanged(w: Int, h: Int) {
+        glThread!!.onWindowResize(w, h)
+    }
+
+    override fun onVisibilityChanged(changedView: View, visibility: Int) {
+        if (!isInEditMode) {
+            if (visibility == View.GONE || visibility == View.INVISIBLE) {
+                onPause()
+            } else {
+                onResume()
+            }
+        }
+        super.onVisibilityChanged(changedView, visibility)
+    }
+
+    /**
+     * This method is used as part of the View class and is not normally
+     * called or subclassed by clients of TextureView.
+     */
+    override fun onAttachedToWindow() {
+        super.onAttachedToWindow()
+        if (LOG_ATTACH_DETACH) {
+            Log.d(TAG, "onAttachedToWindow reattach =$detached")
+        }
+        if (detached && rendererDelegate != null) {
+            var renderMode = ISurface.RENDERMODE_CONTINUOUSLY
+            if (glThread != null) {
+                renderMode = glThread!!.renderMode
+            }
+            glThread = GLThread(thisWeakRef)
+            if (renderMode != ISurface.RENDERMODE_CONTINUOUSLY) {
+                glThread!!.renderMode = renderMode
+            }
+            glThread!!.start()
+        }
+        detached = false
+    }
+
+    override fun onDetachedFromWindow() {
+        if (LOG_ATTACH_DETACH) {
+            Log.v(TAG, "onDetachedFromWindow")
+        }
+        rendererDelegate!!.mRenderer.onRenderSurfaceDestroyed(null)
+        if (glThread != null) {
```

Comment:
glThread?.requestExitAndWait()

Reply:
I changed all ` != null` comparisons to `?` and with blocks to `xx?.let {}`